app/controllers/users/confirmations_controller.rb

```diff
@@ -112,7 +112,12 @@ def after_confirmation_path_for(resource)
     end
 
     def process_confirmed_user
-      analytics.track_event('Email changed and confirmed', @confirmable)
+      if @confirmable.previous_changes.key?('email')
+        analytics.track_event('Email change confirmed', @confirmable)
+        create_user_event(:email_changed, @confirmable)
+      else
+        analytics.track_event('Email address confirmed', @confirmable)
+      end
```

Comment:
I could be wrong, but I'm pretty sure this method will only be called when an existing user changes their email and confirms it, so there's no need for the conditional.

On a higher level, what do you think about calling Event.create from within the Analytics class `track_event` method, as opposed to adding an additional method in each controller. The idea is that the Analytics class will fire off multiple things every time we want to track an event. Currently, it sends the event to Google Analytics, writes to the Rails log, and tracks the event via ahoy. If we want to fire off a new event, we would add that to the Analytics class. That way, if we decide we longer want to execute a particular command, we can simply remove one line from the Analytics class, as opposed to each occurrence of the call over separate controllers.